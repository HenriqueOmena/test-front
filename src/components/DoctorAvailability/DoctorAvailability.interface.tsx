export interface IDoctorAvailability {
  data: { start: string }[];
  id: string;
  name: string;
}
