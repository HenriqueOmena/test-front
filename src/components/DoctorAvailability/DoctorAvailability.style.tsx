import { makeStyles } from "@material-ui/core";

const DoctorAvailabilityStyle = makeStyles(() => ({
  availability: {
    marginTop: 10
  },
  availabilityItem: {
    display: "flex",
    justifyContent: "space-between",
    alignItems: "center",
    marginTop: 5
  }
}));

export default DoctorAvailabilityStyle;
