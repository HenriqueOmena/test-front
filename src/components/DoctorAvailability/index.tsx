export { default as DoctorAvailability } from "./DoctorAvailability";
