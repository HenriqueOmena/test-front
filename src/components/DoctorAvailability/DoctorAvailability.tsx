import { Button } from "@material-ui/core";
import { format } from "date-fns";
import React from "react";
import { Link } from "react-router-dom";
import { IDoctorAvailability } from "./DoctorAvailability.interface";
import DoctorAvailabilityStyle from "./DoctorAvailability.style";

const DoctorAvailability = ({ data, id, name }: IDoctorAvailability) => {
  const classes = DoctorAvailabilityStyle();
  // const [startDate, setStartDate] = useState<string>();

  // FIXME: i had to ignored that cause im getting error with CORS from server i try to change proxy configs but without success
  // const { apiData, isLoading, execute } = useRequest("POST", `bookings`, false, {
  //   date: startDate,
  //   doctorId
  // });

  const generateDate = (date: string) => {
    const newDate = new Date(date);
    return format(newDate, "yyyy / MM / dd - HH:mm:ss");
  };

  const makeAppointment = (date: string) => {
    // setStartDate(date);
    // execute();
  };

  return (
    <div className={classes.availability}>
      <ul>
        {data.map(({ start }, index) => (
          <li key={index} className={classes.availabilityItem}>
            {generateDate(start)} :
            <Link
              to={{
                pathname: "/bookings",
                state: { start, name },
              }}
            >
              <Button variant="contained" color="primary" onClick={() => makeAppointment(start)}>
                To Schedule
              </Button>
            </Link>
          </li>
        ))}
      </ul>
    </div>
  );
};

export default DoctorAvailability;
