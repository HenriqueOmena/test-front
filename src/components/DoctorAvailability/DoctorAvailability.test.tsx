import { fireEvent, render } from "@testing-library/react";
import React from "react";
import { MemoryRouter } from "react-router-dom";
import DoctorAvailability from "./DoctorAvailability";

describe("Component - DoctorAvailability", () => {
  const propMock = {
    data: [{ start: "2021-08-26T05:49:16.334Z" }],
    id: "anyId",
    name: "anyName"
  };

  let component = (
    <MemoryRouter>
      <DoctorAvailability {...propMock} />;
    </MemoryRouter>
  );

  it("Should Match SnapShot", () => {
    expect(component).toMatchSnapshot();
  });

  it("Should render component", async () => {
    render(component);
  });

  it("Should click at to Schedule button", async () => {
    const { getByText } = render(component);
    const button = getByText("To Schedule");
    fireEvent.click(button);
  });
});
