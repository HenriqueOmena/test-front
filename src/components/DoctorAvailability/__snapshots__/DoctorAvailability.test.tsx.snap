// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component - DoctorAvailability Should Match SnapShot 1`] = `
<MemoryRouter>
  <DoctorAvailability
    data={
      Array [
        Object {
          "start": "2021-08-26T05:49:16.334Z",
        },
      ]
    }
    id="anyId"
    name="anyName"
  />
  ;
</MemoryRouter>
`;
